src/app/applications/applications.controller.app.js

```diff
@@ -41,7 +42,12 @@
 
     function deleteApplication() {
       rClient.deleteApplication($routeParams.appName).then(function() {
-        $window.location = '#/';
+          $window.location = '#/';
+      });
+    }
+    function uploadIcon(appName) {
+      rClient.uploadIcon(appName).then(function() {
+          $window.location = '#/';
```

Comment:
This means you redirect the browser to the welcome page.
Instead, you should redirect to the application's detailed page.